packages/connection-manager/src/ConnectionQueueItem.js

```diff
@@ -0,0 +1,50 @@
+import { AbstractConnection } from "connections";
+
+export const DEFAULT_PRIORITY = 0;
+export const MINIMUM_PRIORITY = DEFAULT_PRIORITY;
+
+export default class ConnectionQueueItem {
+  /**
+   * @param {AbstractConnection} connection – given connection
+   * @param {int} [priority=DEFAULT_PRIORITY] – given priority
+   */
+  constructor(connection, priority = DEFAULT_PRIORITY) {
+    if (!(connection instanceof AbstractConnection)) {
+      throw new Error(
+        "Invalid Connection, has to be an instance of AbstractConnection."
+      );
+    }
+
+    if (typeof priority !== "number" || priority < MINIMUM_PRIORITY) {
```

Comment:
Let's use the polyfill for `Number.isInteger(priority)` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger since it's not available in IE and enforce integer, or change the parameter type to generic number.

Reply:
Edit on above comment: The polyfill ships with our babel toolset already.